### DISCUSSION TITLE
Enhance CLAHE - Jit compatible

### DISCUSSION THREAD

**Discussion by milliemcq** [kornia/kornia#2672] [Questions]
Currently trying to compile enhance_clahe, but it's failing to compile due to `_compute_tiles` is there a way around this? It looks like it's supposed to be compilable? 🤔 

```
`RuntimeError: 
undefined value _compute_tiles:
  File "site-packages/kornia/utils/image.py", line 362
    # img_padded: torch.Tensor  # B x C x H' x W'  # not supported by JIT
    # the size of the tiles must be even in order to divide them into 4 tiles for the interpolation
    hist_tiles, img_padded = _compute_tiles(imgs, grid_size, True)
                             ~~~~~~~~~~~~~~ <--- HERE
    tile_size: Tuple[int, int] = (hist_tiles.shape[-2], hist_tiles.shape[-1])
    interp_tiles: torch.Tensor = _compute_interpolation_tiles(img_padded, tile_size)  # B x 2GH x 2GW x C x TH/2 x TW/2
equalize_clahe is being compiled since it was called from 'Preprocess.apply_clahe'
  File preprocess.py", line 398
        images_clahe = K.enhance.equalize_clahe(images, clip_limit=5.0, grid_size=(2, 2))
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
Preprocess.apply_clahe is being compiled since it was called from 'Preprocess.forward'
  File "x", line 275
            inputs = self.apply_clahe(inputs)
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE`
```
pytorch version = 2.0.1
kornia = 0.6.9

**Comment by edgarriba** [comment]
after looking at it, seems that at some point that was supported: https://github.com/kornia/kornia/blame/master/test/enhance/test_equalization.py#L85

but after adding https://github.com/kornia/kornia/blob/master/kornia/enhance/equalization.py#L309 -- we dropped support. I'd be in favour of restoring the torchscript/torchcompile support here

@milliemcq Are you willing to contribute with that ? /cc @johnnv1 @lferraz

**Comment by milliemcq** [comment]
Sure, so the decorator is all the problem here? Can I just remove? Appreciate it wouldn't support dynamic reshaping but most could do themselves?

**Reply by johnnv1** [reply]
ideally, we wouldn't change the current feature, what we're thinking of for another case of this, would be just to implement within decorator functionality inside the function

**Reply by edgarriba** [reply]
yes, remove the decorator and possibly need to add inside the function the behaviour to .view(-1, ...) -> compute -> .view_as(input)
i'd suggest to run `pytest test/enhance/test_equalization.py` to make sure we make backward compatible. Uncomment the test there too, possibly need to be updated to something like this: https://github.com/kornia/kornia/blob/master/test/color/test_gray.py#L206 @johnnv1  can guide
